src/config.c

```diff
@@ -2709,6 +2709,14 @@ static int setConfigDirOption(standardConfig *config, sds *argv, int argc, const
         *err = "dir can't be empty";
         return 0;
     }
+    if (hasActiveChildProcess() && server.child_type == CHILD_TYPE_AOF) {
+        *err = "it is unsafe to modify dir when a AOF rewrite is running";
+        return 0;
+    }
+    if (hasActiveChildProcess() && server.logfile[0] != '\0') {
```

Comment:
```suggestion
    if (hasActiveChildProcess()) {
```
Maybe we also do this just so that we don't have to worry about this looking forward?

Reply:
yes. Make sense to me

Reply:
i will fix the test soon